src/main/java/net/minecraftforge/unification/UnificationManager.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.unification;
+
+import javax.annotation.Nullable;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.oredict.OreDictionary;
+
+/**
+ * The unification manager is the main access point for getting instances of:
+ * {@link UnificationMaterial}, {@link UnificationForm}, and {@link UnificationRegistry}.
+ */
+public class UnificationManager
+{
+    private static final Map<String, UnificationMaterial> MATERIALS = new HashMap<>();
+    private static final Map<String, UnificationForm> FORMS = new HashMap<>();
+    @Nullable
+    private static UnificationRegistry<ItemStack> ITEM_STACK_UNIFICATION_REGISTRY;
+    @Nullable
```

Comment:
Is there any annotation like `@LazyInit`, which are more clear?

Reply:
No, unfortunately. 
I use this pattern all the time though. It's `@Nullable` on the private method and `@Nonnull` on the public getter.